pkg/build/legacy/parser/deffile.go

```diff
@@ -0,0 +1,413 @@
+		if err != nil && err != bufio.ErrFinalToken {
+			return 0, nil, err
+		} else if line == nil { // If ScanLines returns a nil line, it needs more data. Send req for more data
+			return 0, nil, nil // Returning 0, nil, nil requests Scanner.Scan() method find more data or EOF
+		}
+
+		_, word, err := bufio.ScanWords(line, true) // Tokenize the line into words
+		if err != nil && err != bufio.ErrFinalToken {
+			return 0, nil, err
+		}
+
+		// Check if the first word starts with % sign
+		if word != nil && word[0] == '%' {
+			// If the word starts with %, it's a section identifier
+
+			// We no longer check if the word is a valid section identifier here, since we want to move to
+			// a more modular approach where we can parse arbitrary sections
+			if inSection {
+				// Here we found the end of the section
+				return advance, retbuf.Bytes(), nil
+			} else if advance == 0 {
+				// When advance == 0 and we found a section identifier, that means we have already
+				// parsed the header out and left the % as the first character in the data. This means
+				// we can now parse into sections.
+				retbuf.Write(line)
+				retbuf.WriteString("\n")
+				inSection = true
+			} else {
+				// When advance != 0, that means we found the start of a section but there is
+				// data before it. We return the data up to the first % and that is the header
+				return advance, retbuf.Bytes(), nil
+			}
+		} else {
+			// This line is not a section identifier
+			retbuf.Write(line)
+			retbuf.WriteString("\n")
+		}
+
+		// Shift the advance retval to the next line
+		advance += a
+		if a == 0 {
+			break
+		}
+	}
+
+	if !atEOF {
+		return 0, nil, nil
+	}
+
+	return advance, retbuf.Bytes(), nil
+
+}
+
+func getSectionName(line string) string {
+	// trim % prefix on section name
+	line = strings.TrimLeft(line, "%")
+	lineSplit := strings.SplitN(strings.ToLower(line), " ", 2)
+
+	return lineSplit[0]
+}
+
+// parseTokenSection splits the token into maximum 2 strings separated by a newline,
+// and then inserts the section into the sections map
+//
+func parseTokenSection(tok string, sections map[string]string) error {
+	split := strings.SplitN(tok, "\n", 2)
+	if len(split) != 2 {
+		return fmt.Errorf("Section %v: Could not be split into section name and body", split[0])
+	}
+
+	key := getSectionName(split[0])
+	if appSections[key] {
+		sectionSplit := strings.SplitN(strings.TrimLeft(split[0], "%"), " ", 3)
+		if len(sectionSplit) < 2 {
+			return fmt.Errorf("App Section %v: Could not be split into section name and app name", sectionSplit[0])
+		}
+
+		key = strings.Join(sectionSplit[0:2], " ")
+	}
+
+	sections[key] += split[1]
+
+	return nil
+}
+
+func doSections(s *bufio.Scanner, d *types.Definition) error {
+	sectionsMap := make(map[string]string)
+
+	tok := strings.TrimSpace(s.Text())
+
+	// skip initial token parsing if it is empty after trimming whitespace
+	if tok != "" {
+		//check if first thing parsed is a header/comment or just a section
```

Comment:
I would add a space here for consistency:

```suggestion
		// Check if first thing parsed is a header/comment or just a section
```